dspace-spring-rest/src/main/java/org/dspace/app/rest/IdentifierRestController.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ * <p>
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+
+import java.net.URI;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.log4j.Logger;
+import org.atteo.evo.inflector.English;
+
+import org.dspace.app.rest.converter.DSpaceObjectConverter;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.DSpaceObjectRest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.content.DSpaceObject;
+import org.dspace.core.Context;
+import org.dspace.identifier.IdentifierNotFoundException;
+import org.dspace.identifier.IdentifierNotResolvableException;
+import org.dspace.identifier.factory.IdentifierServiceFactory;
+import org.dspace.identifier.service.IdentifierService;
+
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.Link;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/pid")
+public class IdentifierRestController implements InitializingBean {
+
+    private static final Logger log =
+        Logger.getLogger(IdentifierRestController.class);
+
+    @Autowired
+    DiscoverableEndpointsService discoverableEndpointsService;
+
+    @Autowired
+    private HalLinkService halLinkService;
+
+    @Autowired
+    private List<DSpaceObjectConverter> converters;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        List<Link> links = new ArrayList<Link>();
+
+        Link l = new Link("/api", "pid");
+        links.add(l);
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/{prefix}/{suffix}")
+    @SuppressWarnings("unchecked")
+    public void getDSObyIdentifier(@PathVariable String prefix,
```

Comment:
I will prefer to avoid the use of prefix/suffix.
I know that simply using /{identifier} doesn't work, this could help https://stackoverflow.com/questions/3686808/spring-3-requestmapping-get-path-value
it is spring 3, we are on 4 but...

Reply:
I have tried using value="/**", but the endpoint is not found.  I have also tried using a constant, but again, the slash in the handle seems to present a problem.  I will keep looking, but if you have any ideas let me know!!